File: 04.Objects-And-Classes-More-Exercise/catalogue.js
function catalogue(arrayOfProducts) {
    arrayOfProducts = arrayOfProducts.sort()
    let arrayOfFirstLetters = []
    
    for (const element of arrayOfProducts) {
        if (!arrayOfFirstLetters.includes(element[0])) {
            arrayOfFirstLetters.push(element[0])
        }
    }

    for (const letter of arrayOfFirstLetters) {
        console.log(letter)

        for (const element of arrayOfProducts) {
            if (letter === element[0]) {
                let [productName, productPrice] = element.split(" : ")
                console.log("  " + productName + ": " + productPrice)
            }
        }
    }
}

catalogue([
    'Appricot : 20.4',
    'Fridge : 1500',
    'TV : 1499',
    'Deodorant : 10',
    'Boiler : 300',
    'Apple : 1.25',
    'Anti-Bug Spray : 15',
    'T-Shirt : 10'
    ]
    )

File: 04.Objects-And-Classes-More-Exercise/catalogue.py
def catalogue(list_of_products):
    list_of_products = sorted(list_of_products)
    list_of_first_letters = []

    for element in list_of_products:
        if element[0] not in list_of_first_letters:
            list_of_first_letters.append(element[0])

    for letter in list_of_first_letters:
        print(letter)

        for element in list_of_products:
            if letter == element[0]:
                product_name, product_price = element.split(" : ")
                print("  " + product_name + ": " + product_price)


catalogue([
    'Appricot : 20.4',
    'Fridge : 1500',
    'TV : 1499',
    'Deodorant : 10',
    'Boiler : 300',
    'Apple : 1.25',
    'Anti-Bug Spray : 15',
    'T-Shirt : 10'
    ]
    )